### PR TITLE
Mini Cart Block: Remove not necessary div #5419

### LINKED ISSUE #5419: The `div` wrapper of RawHTML isn't removed on the front e...
The &#x60;div&#x60; wrapper of RawHTML isn&#x27;t removed on the front end. */ }

---
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/8d08c6056afede0f97ac204e6add679bc8d6e0d1/assets/js/blocks/cart-checkout/mini-cart/block.tsx#L197-L200
---
###### :rocket: This issue was generated by the [automations bot](https://github.com/woocommerce/automations) based on a `todo` comment in 8d08c6056afede0f97ac204e6add679bc8d6e0d1 when #5386 was merged. cc @dinhtungdu

### PR DESCRIPTION
**Pull request by gigitux**
This PR removes a not necessary div to prevent not necessary nested DOM structure.

Fixes #5419

### Manual Testing

How to test the changes in this Pull Request:

Check out this branch:

1. Using Storefront, add the Mini Cart block as widget.
2. View the page and inspect elements using the browser's dev tools.
3. Click on the mini-cart to open the mini-cart-contents drawer.
4. Find the `wc-block-mini-cart__template-part` class element.
5. Check that the direct child is a div with `wp-block-woocommerce-mini-cart-contents` class element.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/mini-cart/block.tsx [hunk 1]
```diff
@@ -193,10 +192,8 @@ const MiniCartBlock = ( {
 				<div
 					className="wc-block-mini-cart__template-part"
 					ref={ contentsRef }
-				>
-					{ /* @todo The `div` wrapper of RawHTML isn't removed on the front end. */ }
-					<RawHTML>{ contents }</RawHTML>
-				</div>
+					dangerouslySetInnerHTML={ { __html: contents } }
```
Comment (opr) [inline assets/js/blocks/cart-checkout/mini-cart/block.tsx:195]: How do we know this is safe, can we ensure that `contents` contains nothing malicious?

Comment (gigitux) [reply assets/js/blocks/cart-checkout/mini-cart/block.tsx:195]: image

This is the source code of `RawHTML`, there isn't any check about checking the sanity of the content.

This content is loaded from the `templates` folder and [parsed from the backend](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/66e61cd502fd83d28338b3fcfec340c18fb8fded/src/Utils/BlockTemplateUtils.php#L171).

So I guess it is a pretty safe approach. What do you think?

Comment (opr) [reply assets/js/blocks/cart-checkout/mini-cart/block.tsx:195]: Hey! Thanks for clarifying. Yep it seems fine, the concern of adding malicious scripts in a template is pretty low as a merchant would have to modify their template files, something we can't really stop.

LGTM 👍🏼

Comment (gigitux) [reply assets/js/blocks/cart-checkout/mini-cart/block.tsx:195]: Thanks!

**Review by opr — APPROVED** [review]
🥳 Works fine for me! Let's 🚢 it!

**Pull request merged**